internal/service/ivs/recording_configuration.go

```diff
@@ -0,0 +1,334 @@
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-z0-9-.]{3,63}$`), "must contain only lowercase alphanumeric characters, hyphen, or dot, and between 3 and 63 characters"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9-_]{0,128}$`), "must contain only alphanumeric characters, hyphen, or underscore, and at most 128 characters"),
+			},
+			"recording_reconnect_window_seconds": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 300),
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags":     tftags.TagsSchemaForceNew(),
+			"tags_all": tftags.TagsSchemaComputed(),
+			"thumbnail_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"recording_mode": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringInSlice(ivs.RecordingMode_Values(), false),
+						},
+						"target_interval_seconds": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.IntBetween(5, 60),
+						},
+					},
+				},
+			},
+		},
+
+		CustomizeDiff: verify.SetTagsDiff,
+	}
+}
+
+const (
+	ResNameRecordingConfiguration = "Recording Configuration"
+)
+
+func resourceRecordingConfigurationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).IVSConn
+
+	in := &ivs.CreateRecordingConfigurationInput{
+		DestinationConfiguration: expandDestinationConfiguration(d.Get("destination_configuration").([]interface{})),
+	}
+
+	if v, ok := d.GetOk("name"); ok {
+		in.Name = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("recording_reconnect_window_seconds"); ok {
+		in.RecordingReconnectWindowSeconds = aws.Int64(int64(v.(int)))
+	}
+
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	if len(tags) > 0 {
+		in.Tags = Tags(tags.IgnoreAWS())
+	}
+
+	if v, ok := d.GetOk("thumbnail_configuration"); ok {
+		in.ThumbnailConfiguration = expandThumbnailConfiguration(v.([]interface{}))
+
+		if aws.StringValue(in.ThumbnailConfiguration.RecordingMode) == ivs.RecordingModeDisabled && in.ThumbnailConfiguration.TargetIntervalSeconds != nil {
+			return diag.Errorf("thumbnail configuration target interval cannot be set if recording_mode is \"DISABLED\"")
+		}
```

Comment:
If `TargetIntervalSeconds` is set when `RecordingMode` is `"DISABLED"`, the IVS service responds with a generic error:

> Request member: recordingConfiguration.thumbnailConfiguration failed to satisfy the constraints

I thought it would be pertinent to have a more detailed error check here. Is this acceptable? Since the schema cannot have a `ValidateFunc` on a `TypeList`, I believe it would have to go here.

Reply:
👍